### PR TITLE
Forward authentication calls to downstream provider

### PR DESCRIPTION
**Pull request by wwajerowicz**
Part of #7

### CODE REVIEW

File: modules/service/src/main/scala/ch/epfl/bluebrain/nexus/iam/service/routes/AuthRoutes.scala [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+          headerValueByType[Authorization](()) { authHeader =>
+            complete(cl(Get(config.userinfoEndpoint).addHeader(authHeader)))
+          }
+        }
+    }
+}
+
+object AuthRoutes {
+  // $COVERAGE-OFF$
+  def apply(config: OidcConfig)(implicit cl: UntypedHttpClient[Future]): AuthRoutes = {
```
Comment (umbreak) [inline modules/service/src/main/scala/ch/epfl/bluebrain/nexus/iam/service/routes/AuthRoutes.scala]: Missing docs

File: modules/service/src/test/scala/ch/epfl/bluebrain/nexus/iam/service/routes/AuthRoutesSpec.scala [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+import ch.epfl.bluebrain.nexus.iam.service.config.AppConfig.OidcConfig
+import org.mockito.Mockito
+import org.mockito.Mockito._
+import org.scalatest.mockito.MockitoSugar
+import org.scalatest.prop.TableDrivenPropertyChecks
+import org.scalatest.{BeforeAndAfter, Matchers, WordSpecLike}
+
+import scala.concurrent.Future
+
+class AuthRoutesSpec  extends WordSpecLike
```
Comment (umbreak) [inline modules/service/src/test/scala/ch/epfl/bluebrain/nexus/iam/service/routes/AuthRoutesSpec.scala]: Needed to add test cases for failures I guess. What happen when those endpoints you are forwarding in the routes don't respond?
What happens when a client hits the path /oauth2/userinfo without the header of type Authorization ?

Mostly all of it is handled right now by Akka HTTP. However we probably need to add our own RejectionHandling on it.

File: modules/service/src/main/scala/ch/epfl/bluebrain/nexus/iam/service/Main.scala [hunk 3]
```diff
@@ -43,6 +45,7 @@ object Main {
 
     val logger           = Logging(as, getClass)
     val sourcingSettings = SourcingAkkaSettings(journalPluginId = appConfig.persistence.queryJournalPlugin)
+    implicit val cl: UntypedHttpClient[Future] = HttpClient.akkaHttpClient
```
Comment (bogdanromanx) [inline modules/service/src/main/scala/ch/epfl/bluebrain/nexus/iam/service/Main.scala]: move this next to the other implicit values (line 45)

File: modules/service/src/main/resources/app.conf [hunk 4]
```diff
@@ -73,16 +73,11 @@ app {
...
-    # The platform client id used for authenticating to the server
-    client-id = ${?OIDC_CLIENT_ID}
-    # The platform client secret used for authenticating to the server
-    client-secret = ${?OIDC_CLIENT_SECRET}
-    # The authorization scope
-    scope = ["openid"]
-    # This application's token uri
-    token-uri = ${app.http.public-uri}"/"${app.http.prefix}"/oauth2/token"
-    token-uri = ${?TOKEN_URI}
+    authorize-endpoint = "http://localhost:8080/oauth2/authorize"
```
Comment (bogdanromanx) [inline modules/service/src/main/resources/app.conf]: please add some comments on the values that need to be set on these configuration elements

**Pull request merged**